core/frontend/src/render/primitives/geometry/GeometryPrimitives.ts

```diff
@@ -270,6 +260,9 @@ class SolidPrimitiveGeometry extends Geometry {
   protected _getPolyfaces(opts: StrokeOptions): PolyfacePrimitiveList {
     const builder = PolyfaceBuilder.create(opts);
     builder.addGeometryQuery(this._primitive);
-    return new PolyfacePrimitiveList(PolyfacePrimitive.create(this.displayParams, builder.claimPolyface()));
+    const pf = builder.claimPolyface();
+    if (!this.transform.isIdentity)
+      pf.tryTransformInPlace(this.transform);

```

Comment:
Would be cheaper to transform the solid primitive before facetting than to transform the polyface.